fix flow-isolation diagnostic message crash

kavon opened the pull request:
resolves rdar://88776902

kavon commented:
@swift-ci please test and merge

kavon commented:
@swift-ci please test

kavon commented on lib/SILOptimizer/Mandatory/FlowIsolation.cpp:
```diff
@@ -367,6 +367,14 @@ static StringRef verbForInvoking(ValueDecl *value) {
+  if (nameable->hasName())
+    return nameable->getName();
+
+  return DeclName(Identifier::getFromOpaquePointer((void*)"_"));
```
I couldn't find that one, but after around I found `ASTContext::getIdentifier(StringRef)` so I'll use that. Thanks 😄

slavapestov replied:
That's what I meant, thanks!

kavon commented:
@swift-ci please smoke test and merge

kavon committed:
fix busted assumption about decl context

When emitting the note to point out what introduced
nonisolation, the code building the message assumed that
`DeclContext::getDecl` will not return null, when it can
if it's a closure expression.

fixes rdar://88776902

kavon commented:
I figured out what went wrong; the decl name was not it.

@swift-ci please smoke test and merge

kavon committed:
beef up the test

kavon commented:
@swift-ci please smoke test and merge
